File: python/tests/simple_usage.py
from eth_account import Account
import dcn


def main() -> None:
    sdk = dcn.Client()

    version = sdk.version()
    print(version["version"])
    print(version["build_timestamp"])

    connector = sdk.connector_get("pitch")
    print(connector["format_hash"])

    feed = sdk.feed(limit=5, include_unfinalized=True)
    print([item["payload"]["name"] for item in feed["items"]])

    account = Account.create()
    sdk.login_with_account(account)

    output = sdk.execute(
        "pitch",
        8,
        {"0": {"start_point": 12, "transformation_shift": 3}},
    )
    print(output[0]["path"])
    print(output[0]["data"])


if __name__ == "__main__":
    main()

File: js/test/simple_usage.ts
import { DcnClient } from '@hypermusic-ai/dcn-js';
import { Wallet } from 'ethers';

async function main() {
    const sdk = new DcnClient();

    const version = await sdk.version();
    console.log('API:', version.version, version.build_timestamp);

    const connector = await sdk.connectorGet('pitch');
    console.log('pitch format:', connector.format_hash);

    const feed = await sdk.feed({ limit: 5, includeUnfinalized: true });
    console.log('feed items:', feed.items.map((item) => item.payload.name));

    const wallet = Wallet.createRandom();
    await sdk.loginWithWallet(wallet);

    const output = await sdk.execute('pitch', 8, {
        '0': { start_point: 12, transformation_shift: 3 },
    });
    console.log('execute pitch result:', output);
}

main();
